ipykernel/control.py

```diff
@@ -1,25 +1,24 @@
-from threading import Thread
-import zmq
-if zmq.pyzmq_version_info() >= (17, 0):
-    from tornado.ioloop import IOLoop
-else:
-    # deprecated since pyzmq 17
-    from zmq.eventloop.ioloop import IOLoop
+from threading import Thread, Event
+import asyncio
 
 
 class ControlThread(Thread):
 
     def __init__(self, **kwargs):
         Thread.__init__(self, name="Control", **kwargs)
-        self.io_loop = IOLoop(make_current=False)
         self.pydev_do_not_trace = True
         self.is_pydev_daemon_thread = True
+        self.io_loop = None
+        self.loop_ready = Event()
+        self.start()
```

Comment:
It is not a good idea to start a thread in the init (people that want to extend this class will be very annoyed by that). We should respect the behavior of built-in Thread and we should avoid changing the behavior for backward compatible reason.

Reply:
At some point I had to put `start` in the `__init__` but it seems that it now works as before, with a separate `start` method. I'll push this change, thanks!

Reply:
At some point I had to put `start` in the `__init__` but it seems that it now works as before, with a separate `start` method. I'll push this change, thanks!